pkg/services/auth/jwt.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+var SecretKey string
+
+const ExpirationTime = time.Minute * 15
+
+// Claims is a custom JWT claims that contains some token information
+type Claims struct {
+	jwt.StandardClaims
+	Provider      string `json:"provider"`
+	ProviderToken string `json:"provider_token"`
+}
+
+// Generate generates and signs a new token
+func Generate(secretKey string, expirationTime time.Duration, providerName, providerToken string) (string, error) {
```

Comment:
We have a typed provider name now. I believe its `gitproviders.GitProviderNameGithub`